app/util/analytics.py

```diff
@@ -121,46 +118,88 @@ def get_actual_run_time(self):
         run_time_start_finish = self.get_duration_by_start_finish_strings()
         return run_time_bzt if run_time_bzt else run_time_start_finish
 
-    def get_actual_test_count(self):
-        jmeter_test = ' jmeter_'
-        selenium_test = ' selenium_'
-        for line in self.bzt_log_file:
-            if jmeter_test in line:
-                self.jmeter_test_count = self.jmeter_test_count + 1
-            elif selenium_test in line:
-                self.selenium_test_count = self.selenium_test_count + 1
+    def get_test_count_by_type(self, tests_type, log):
+            trigger = f' {tests_type}_'
+            test_search_regx = ""
+            if tests_type == 'jmeter':
+                test_search_regx = JMETER_TEST_REGX
+            elif tests_type == 'selenium':
+                test_search_regx = SELENIUM_TEST_REGX
+            tests = {}
+            for line in log:
+                if trigger in line and ('FAIL' in line or 'OK' in line):
+                    test_name = re.findall(test_search_regx, line)[0]
+                    test_rate = float(''.join(re.findall(SUCCESS_TEST_RATE_REGX, line))[:-1])
+                    if test_name not in tests:
+                        tests[test_name] = test_rate
+            return tests
+
+    def get_success_count_from_tests(self, tests):
+        success_test_count = 0
+        for _, success_rate in tests.items():
+            if success_rate >= SUCCESS_TEST_RATE:
+                success_test_count = success_test_count + 1
+        return success_test_count
+
+
+    def set_actual_test_count(self):
+        TEST_RESULTS_START_STRING = 'Request label stats:'
+        res_string_idx = [index for index, value in enumerate(self.bzt_log_file) if TEST_RESULTS_START_STRING in value]
+        # Cut bzt.log from the 'Request label stats:' string to the end
+        if res_string_idx:
+            res_string_idx = int(res_string_idx[0])
```

Comment:
The list should already contain `int` value of `index`. It would be better to calc it on line 147.